quantumblacklabs/kedro-viz
# [KED-2849] & [KED-2851] Set up GraphQL mock server

## Description

related tickets: [KED-2849](https://jira.quantumblack.com/browse/KED-2849) & [KED-2851](https://jira.quantumblack.com/browse/KED-2851)

This PR includes the work of setting up the mock server, as well as the apollo client for the consumption of the data. I have also hooked the sidebar component up with the apollo client to consume graphql data. 

![image](https://user-images.githubusercontent.com/27775658/138490321-a4ca6122-fe63-46b6-9ffc-859775b004d4.png)


## Development notes

- Set up of apollo client ( `config.js` )
- Set up of mocks to mock data types according to our schema
- Set up of queries for the consumption of graphql data ( this is set up according to the data structure and needs by the FE components) - this will be where all future queries live in the codebase
- Set up of the Schema - this will be the schema that is followed by both FE and BE
- Set up of the Provider that links up to the client. I have currently set up 2 providers, the `ApolloProvider` for the consumption of graphql data served from the BE, an the `MockedProvider` for the consumption of the mocked data
- updated the experiment wrapper to consume data from the Provider
- Updated the sidebar components ( `node-list-card` and `sidebar`) to take in the mocked data
- updates to the `node-list-card` component with data types

## QA notes

There are no additional 'tests' for this PR as this is mostly setup / config work, and there are not really any need to test the functionality of Apollo / faker.js. ( both are well battle-tested libraries)

The best way to test this is to check out the branch, and log the data returned from the query. You will also see the random data populating the sidebar. 

## Checklist


## Legal notice


- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
